diambra/arena/stable_baselines3/sb3_utils.py

```diff
@@ -0,0 +1,52 @@
+from stable_baselines3.common.callbacks import BaseCallback
+import cv2
+import os
+import time
+import json
+import numpy as np
+
+# Linear scheduler for RL agent parameters
+def linear_schedule(initial_value, final_value=0.0):
+    """
+    Linear learning rate schedule.
+    :param initial_value: (float or str)
+    :return: (function)
+    """
+    if isinstance(initial_value, str):
+        initial_value = float(initial_value)
+        final_value = float(final_value)
+        assert (initial_value > 0.0), "linear_schedule work only with positive decreasing values"
+
+    def func(progress):
+        """
+        Progress will decrease from 1 (beginning) to 0
+        :param progress: (float)
+        :return: (float)
+        """
+        return final_value + progress * (initial_value - final_value)
+
+    return func
+
+# AutoSave Callback
+class AutoSave(BaseCallback):
+    """
+    Callback for saving a model, it is saved every ``check_freq`` steps
+
+    :param check_freq: (int)
+    :param save_path: (str) Path to the folder where the model will be saved.
+    :param verbose: (int)
+    """
+    def __init__(self, check_freq: int, num_envs: int, save_path: str, verbose=1):
+        super(AutoSave, self).__init__(verbose)
+        self.check_freq = int(check_freq / num_envs)
+        self.num_envs = num_envs
+        self.save_path_base = save_path + 'autoSave_'
```

Comment:
Any reason this is camel case? I'd go with autosave.
Also, this doesn't ensure save_path as a trailing slash. I think better would be to make save_path type Path (https://docs.python.org/3/library/pathlib.html#module-pathlib) and then use / to join

Reply:
of course, I'll update this thanks for noting